connectors/sources/microsoft_teams.py

```diff
@@ -0,0 +1,1316 @@
+        child = await self.get_user_drive_root_children(drive_id=drive.get("id"))
+        async for attachment in self.scroll(
+            url=URLS[UserEndpointName.ATTACHMENT.value].format(
+                base_url=BASE_URL,
+                drive_id=drive.get("id"),
+                child_id=child.get("id"),
+                attachment_name=url_encode(attachment_name).replace("'", "''"),
+            ),
+        ):
+            yield attachment
+
+    async def get_calendars(self, user_id):
+        async for events in self.scroll(
+            url=URLS[UserEndpointName.MEETING.value].format(
+                base_url=BASE_URL, user_id=user_id
+            ),
+        ):
+            yield events
+
+    async def download_item(self, url, async_buffer):
+        await self.pipe(url=url, stream=async_buffer)
+
+    async def get_teams(self):
+        async for teams in self.scroll(
+            url=URLS[TeamEndpointName.TEAMS.value].format(base_url=BASE_URL)
+        ):
+            yield teams
+
+    async def get_team_channels(self, team_id):
+        async for channels in self.scroll(
+            url=URLS[TeamEndpointName.CHANNEL.value].format(
+                base_url=BASE_URL, team_id=team_id
+            )
+        ):
+            yield channels
+
+    async def get_channel_tabs(self, team_id, channel_id):
+        async for channel_tabs in self.scroll(
+            url=URLS[TeamEndpointName.TAB.value].format(
+                base_url=BASE_URL, team_id=team_id, channel_id=channel_id
+            )
+        ):
+            yield channel_tabs
+
+    async def get_channel_messages(self, team_id, channel_id):
+        async for channel_messages in self.scroll(
+            url=URLS[TeamEndpointName.MESSAGE.value].format(
+                base_url=BASE_URL, team_id=team_id, channel_id=channel_id
+            )
+        ):
+            yield channel_messages
+
+    async def get_channel_file(self, team_id, channel_id):
+        file = await self.fetch(
+            url=URLS[TeamEndpointName.FILE.value].format(
+                base_url=BASE_URL, team_id=team_id, channel_id=channel_id
+            )
+        )
+        return file
+
+    async def get_channel_drive_childrens(self, drive_id, item_id):
+        async for root_childrens in self.scroll(
+            url=URLS[TeamEndpointName.ROOT_DRIVE_CHILDREN.value].format(
+                base_url=BASE_URL, drive_id=drive_id, item_id=item_id
+            )
+        ):
+            for child in root_childrens:
+                if child.get("folder"):
+                    async for documents in self.get_channel_drive_childrens(  # pyright: ignore
+                        drive_id=drive_id, item_id=child["id"]
+                    ):
+                        yield documents
+                yield child
+
+    async def close(self):
+        self._sleeps.cancel()
+        await self._http_session.close()
+
+
+class MicrosoftTeamsDataSource(BaseDataSource):
```

Comment:
I think the `MicrosoftTeamsDataSource` class is taking on too much responsibility. All of the `format_*` funcs in this class can be abstracted to a new class called something like `MicrosoftTeamsFormatter`. It would greatly improve readability.